jack.py

```diff
@@ -1533,6 +1562,65 @@ def get_array(self):
         return np.frombuffer(self.get_buffer(), dtype=np.float32)
 
 
+def _maybe_bytes(data):
+    if type(data) is bytes:
+        data_ptr = _ffi.new("unsigned char[]", len(data))
+        data_ptr[0:len(data)] = data
+        return data_ptr
+    return data
+
+def _returns_bytes(fun):
+    def wrapper(self, cnt):
+        ret_cnt, data = fun(self, cnt)
+        return bytes(_ffi.buffer(data, ret_cnt))
+    return wrapper
+
+class RingBuffer(object):
+    def __init__(self, size=256):
+        ptr = _lib.jack_ringbuffer_create(size + 1)
+        if not ptr:
+            raise JackError(
+                "Could not create ringbuffer of size {0}".format(size))
+        self._ptr = _ffi.gc(ptr, _lib.jack_ringbuffer_free)
+
+    def _read(self, cnt):
+        data = _ffi.new("unsigned char[]", cnt)
+        ret_cnt = _lib.jack_ringbuffer_read(self._ptr, data, cnt)
+        return ret_cnt, data
+    read = _returns_bytes(_read)
+
+    def _peek(self, cnt):
+        data = _ffi.new("unsigned char[]", cnt)
```

Comment:
> Allocating memory just to peek into a ringbuffer ?! Really?

What else shall we do?
We have to pass something to `jack_ringbuffer_peek()` ...
And I guess we'll have to allocate memory for it at some point, I guess we cannot use a fixed-size array (or even a single `char`) on the stack in Python, right?
Shall we add an `out` argument to pass in pre-allocated memory?
Then we would also have to provide a function that allocates memory using `ffi.new()` and returns the cdata object.
Actually we wouldn't need an *additional* argument, since the length can be obtained from the cdata object.

Or is there something more reasonable that we can do?